src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/DynamicMethod.CoreCLR.cs

```diff
@@ -0,0 +1,306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Runtime.Loader;
+using System.Text;
+using System.Threading;
+using static System.Runtime.CompilerServices.RuntimeHelpers;
+
+namespace System.Reflection.Emit
+{
+    public sealed partial class DynamicMethod : MethodInfo
+    {
+        private RuntimeType[] m_parameterTypes;
+        internal IRuntimeMethodInfo? m_methodHandle;
+        private RuntimeType m_returnType;
+        private DynamicILGenerator? m_ilGenerator;
+        private DynamicILInfo? m_DynamicILInfo;
+        private bool m_fInitLocals;
+        private Module m_module;
+        internal bool m_skipVisibility;
+        internal RuntimeType? m_typeOwner; // can be null
+        private MethodInvoker? _invoker;
+        private Signature? _signature;
+
+        // We want the creator of the DynamicMethod to control who has access to the
+        // DynamicMethod (just like we do for delegates). However, a user can get to
+        // the corresponding RTDynamicMethod using Exception.TargetSite, StackFrame.GetMethod, etc.
+        // If we allowed use of RTDynamicMethod, the creator of the DynamicMethod would
+        // not be able to bound access to the DynamicMethod. Hence, we need to ensure that
+        // we do not allow direct use of RTDynamicMethod.
+        private RTDynamicMethod m_dynMethod;
+
+        // needed to keep the object alive during jitting
+        // assigned by the DynamicResolver ctor
+        internal DynamicResolver? m_resolver;
+
+        internal bool m_restrictedSkipVisibility;
```

Comment:
It would be nice to strip the `m_` prefixes while you are on it, so that it conforms with the libraries naming convention.